File: crates/aegis-language/tests/corpora/python/fs_delete.py
# fs_delete.py — positive corpus: filesystem deletion operations.
#
# Expected: four FilesystemDelete operations in source order. The first three
# are single-file removes (no modifiers); `shutil.rmtree` carries the recursive
# modifier. Every operand is a string literal → Known certainty. No execution
# sinks, so no nested payloads. Parses cleanly.
import os
import shutil

# Single-file delete, three spellings.
os.remove("/tmp/a")
os.unlink("/tmp/b")
os.rmdir("/tmp/empty")

# Recursive directory-tree delete.
shutil.rmtree("/tmp/tree")

File: crates/aegis-language/tests/corpora/javascript/fs_delete.js
// Corpus: filesystem deletion (plan Iteration 7, ADR-022 §3).
//
// Four destructive delete call sites spanning the `fs` deletion surface the
// adapter tracks: `unlinkSync` (file), `rmdirSync` (empty dir), and `rmSync`
// with a `recursive: true` options object in both the identifier- and
// string-keyed literal shapes. Each operand is a pure string literal, so
// every operation is `Known`. `rmSync` resolves its `recursive` modifier from
// the options object; `unlinkSync`/`rmdirSync` are never recursive.

fs.unlinkSync("a.txt");
fs.rmdirSync("d1");
fs.rmSync("d2", {recursive: true});
fs.rmSync("d3", {"recursive": true});